src/scancode/cli.py

```diff
@@ -443,6 +446,26 @@ def scan_event(item):
     return files_count, cached_scan.iterate(scans, root_dir)
 
 
+def shorten_filename(filename, max_length=25):
+	"""
+	Return the original filename if it is short enough
+	otherwise return its shortened version 
+	"""
+	if len(filename) <= max_length:
+		shortened_filename = filename
+	else:
+		file_base_name = fileutils.file_base_name(filename)
+		file_extension = fileutils.file_extension(filename)
```

Comment:
use instead  something like 
`base_name, ext = fileutils.splitext(filename)` .... but again you did not receive a filename but a path (though it does not matter to splitext)